lib/active_merchant/billing/gateways/shift4_v2.rb

```diff
@@ -11,6 +11,28 @@ def credit(money, payment, options = {})
         commit('credits', post, options)
       end
 
+      def store(payment_method, options = {})
+        post = case payment_method
+               when CreditCard
+                 cc = {}.tap { |card| add_creditcard(card, payment_method, options) }[:card]
```

Comment:
this line is a little confuse, we could simplify it something like this
```suggestion
                 cc = add_creditcard({}, payment_method, options)[:card]
```